ycmd/completers/typescript/tsserver_completer.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python
+#
+# Copyright (C) 2015 Google Inc.
+#
+# This file is part of YouCompleteMe.
+#
+# YouCompleteMe is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# YouCompleteMe is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with YouCompleteMe.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+import logging
+import os
+import subprocess
+
+from tempfile import NamedTemporaryFile
+
+from ycmd import responses
+from ycmd import utils
+from ycmd.completers.completer import Completer
+
+_logger = logging.getLogger( __name__ )
+
+
+class TSServerCompleter( Completer ):
+  """
+  Completer for TypeScript.
+
+  It uses TSServer which is bundled with TypeScript 1.5
+
+  See the protocol here:
+  https://github.com/Microsoft/TypeScript/blob/2cb0dfd99dc2896958b75e44303d8a7a32e5dc33/src/server/protocol.d.ts
+  """
+
+  def __init__( self, user_options ):
+    super( TSServerCompleter, self ).__init__( user_options )
+
+    binarypath = utils.PathToFirstExistingExecutable( [ 'tsserver' ] )
+    if not binarypath:
+      raise RuntimeError( 'tsserver not found. TypeScript 1.5 or higher is required' )
+
+    self.seq = 0
```

Comment:
Use something more descriptive than `seq`. At a glance, I don't know what this var is for. Also consider adding a comment above it describing what's it used for.